drivers/quota/projectquota.go

```diff
@@ -75,6 +78,25 @@ type Control struct {
 	quotas            map[string]uint32
 }
 
+// Attempt to generate a unigue projectid.  Multiple directories
+// per file system can have quota and they need a group of unigue
+// ids. This function attempts to allocate at least minPorjectIDS(10000)
+// unigue projectids, based on the inode of the basepath.
+func generateUnigueProjectID(path string) (uint32, error) {
+	const minProjectids = 10000
```

Comment:
If this is a global tuning parameter, make it more prominent, at least a top-level constant on the top of the file.

And isn’t it more a `projectIDsAllocatedPerQuotaHome` or something like that? At least it’s more of a “max”ProjectIDs than a “min”projectIDs.

---

AFAICS this doesn’t even _try_ to prevent overlap. Shouldn’t it?

Reply:
I am not sure of the best way to do this. This was just a stab at it, hoping someone would have a silver bullet.